packages/kit/src/pages.ts

```diff
@@ -34,3 +34,32 @@ export function extendRouteRules (route: string, rule: NitroRouteConfig, options
       : defu(opts.routeRules[route], rule)
   }
 }
+
+export interface AddRouteMiddlewareOptions {
+  /**
+   * Override middleware path
+   *
+   * @default false
+   */
+  override?: boolean
+}
+
+export function addRouteMiddleware (input: NuxtMiddleware | NuxtMiddleware[], options: AddRouteMiddlewareOptions = {}) {
+  const nuxt = useNuxt()
+  nuxt.hook('app:resolve', (app) => {
+    const middlewares = Array.isArray(input) ? input : [input]
+
```

Comment:
We can realise a small performance improvement by normalizing immediately rather than every time the app is resolved.

```suggestion
  const middlewares = Array.isArray(input) ? input : [input]
  nuxt.hook('app:resolve', (app) => {
```